index.js

```diff
@@ -56,3 +56,23 @@ exports.handler = async function handler (event, context) {
     await s3.send(putCmd)
   }
 }
+
+/**
+ * Extract an S3Event from the passed SNSEvent.
+ * @param {import('aws-lambda').SNSEvent} snsEvent
+ * @returns {import('aws-lambda').S3Event}
+ */
+function toS3Event (snsEvent) {
+  const s3Event = { Records: [] }
+  for (const snsRec of snsEvent.Records) {
+    try {
+      for (const s3Rec of JSON.parse(snsRec.Sns.Message).Records || []) {
+        if (s3Rec.eventSource !== 'aws:s3') continue
+        s3Event.Records.push(s3Rec)
+      }
+    } catch (err) {
+      console.error(`failed to extract S3Event record from SNSEvent record: ${err.message}`)
```

Comment:
would it be worth logging the `snsRec` so we know what was being processed when the error occurred?